### DISCUSSION TITLE
How to run deployment mode in localhost

### DISCUSSION THREAD

**Discussion by ssageghi** [chatwoot/chatwoot#1535] [Self hosted] [answered]
I try to use the self-hosted version and trying to customize the frontend a little.
Is there a way to run the development or run build so the changes affect?

**Comment by pranavrajs** [comment, marked as answer]
https://www.chatwoot.com/docs/quick-setup#setup-rails-server

**Reply by ssageghi** [reply]
@pranavrajs I always get this error 
`spawn: Exec format error - bin/rails s -p 3000 (Errno::ENOEXEC)`

**Reply by pranavrajs** [reply]
Can you please post the complete logs?

**Comment by g4rry420** [comment]
@pranavrajs I am getting `Invalid login credentials. Please try again` when I try to access the `localhost:3000` using the credentials as described in [here](https://www.chatwoot.com/docs/quick-setup#login-with-credentials). Also in my terminal, I see this error
```
21:38:34 backend.1  | Started POST "/auth/sign_in" for ::1 at 2021-04-08 21:38:34 -0400
21:38:34 backend.1  | Processing by DeviseOverrides::SessionsController#create as HTML
21:38:34 backend.1  |   Parameters: {"email"=>"john@acme.inc", "password"=>"[FILTERED]", "sso_auth_token"=>"[FILTERED]"}
21:38:34 backend.1  | Can't verify CSRF token authenticity.
21:38:34 backend.1  | Completed 401 Unauthorized in 16ms (Views: 0.2ms | ActiveRecord: 2.3ms | Allocations: 2851)
```
Can you please help with it ?

**Reply by pranavrajs** [reply]
@g4rry420 Have you complete this step? https://www.chatwoot.com/docs/quick-setup#setup-rails-server

**Reply by g4rry420** [reply]
@pranavrajs Yes, I did. But while running `bundle exec rake db:reset
`, I saw warning messsage. I am not sure if this is expected or not.
```
Dropped database 'chatwoot_dev'
Dropped database 'chatwoot_test'
WARNING:  could not flush dirty data: Function not implemented
Created database 'chatwoot_dev'
WARNING:  could not flush dirty data: Function not implemented
Created database 'chatwoot_test'
```

**Reply by g4rry420** [reply]
Btw, it worked. I tried it again. Now the credentials is working. Thank You
